repository_cli/cli/records.py

```diff
@@ -175,12 +179,12 @@ def replace_pid(pid: str, pid_identifier: str):
 
     example call:
         invenio repository rdmrecords pids replace -p "fcze8-4vx33"
-        --pid-identifier ' { "doi":
-            { "identifier": "10.48436/fcze8-4vx33", "provider": "unmanaged" }
-        }'
+        --pid-identifier ' { "doi": {
+            "identifier": "10.48436/fcze8-4vx33", "provider": "unmanaged" }}'
     """
-    pid_identifier = json.loads(pid_identifier)
-    if type(pid_identifier) is not dict:
+    try:
+        pid_identifier = json.loads(pid_identifier)
```

Comment:
Please don't change the type of pid_identifier, first it was a string now it should be dict, thats not good.

Reply:
it was always a string and must be valid JSON. nothing but the try/except changed here. this is only to catch invalid JSON formats